File: VB/TreeListCellMerging/MyTreeListOptionsView.vb
' Developer Express Code Central Example:
' How to add a horizontal cells merge capability to an XtraTreeList control
' 
' This sample shows how to add a horizontal cell merge capability to an
' XtraTreeList control.
' 
' You can find sample updates and versions for different programming languages here:
' http://www.devexpress.com/example=E2501

Imports System.ComponentModel
Imports DevExpress.Utils.Controls
Imports DevExpress.XtraTreeList

Namespace TreeListCellMerging
	Public Class MyTreeListOptionsView
		Inherits TreeListOptionsView

		Private allowHorzMerge As Boolean

		Public Sub New()
			MyBase.New()
			allowHorzMerge = False
		End Sub

		<DefaultValue(False)>
		Public Property AllowHorizontalMerge() As Boolean
			Get
				Return allowHorzMerge
			End Get
			Set(ByVal value As Boolean)
				If allowHorzMerge = value Then
					Return
				End If

				allowHorzMerge = value
			End Set
		End Property

		Public Overrides Sub Assign(ByVal options As BaseOptions)
			MyBase.Assign(options)
			Dim optView As MyTreeListOptionsView = TryCast(options, MyTreeListOptionsView)
			If optView Is Nothing Then
				Return
			End If

			Me.allowHorzMerge = optView.AllowHorizontalMerge
		End Sub
	End Class
End Namespace

File: CS/TreeListCellMerging/MyTreeListOptionsView.cs
using DevExpress.Utils.Controls;
using DevExpress.XtraTreeList;
using System.ComponentModel;

namespace TreeListCellMerging
{
    public class MyTreeListOptionsView : TreeListOptionsView
    {
        private bool allowHorzMerge;

        public MyTreeListOptionsView(TreeList treeList)
            : base(treeList)
        {
            allowHorzMerge = false;
        }

        public override void Assign(BaseOptions options)
        {
            base.Assign(options);
            MyTreeListOptionsView optView = options as MyTreeListOptionsView;
            if(optView == null)
                return;

            this.allowHorzMerge = optView.AllowHorizontalMerge;
        }

        [DefaultValue(false)]
        public bool AllowHorizontalMerge {
            get { return allowHorzMerge; }
            set {
                if(allowHorzMerge == value)
                    return;

                allowHorzMerge = value;
            }
        }
    }
}
